accounts-db/src/accounts_index/in_mem_accounts_index.rs

```diff
@@ -2228,4 +2243,44 @@ mod tests {
         assert_eq!(test.slot_list_lock_read_len(), len);
         assert_eq!(len, 2);
     }
+
+    #[test_case(Some(10000);  "with pre-allocation 10000")]
+    #[test_case(Some(20000);  "with pre-allocation 20000")]
+    #[test_case(Some(30000);  "with pre-allocation 30000")]
+    #[test_case(None; "without pre-allocation")]
+    fn test_new_with_num_initial_accounts(num_initial_accounts: Option<usize>) {
+        let config = AccountsIndexConfig::default();
+
+        // Test with different bin counts: 2, 4, 8
+        let bin_counts = [2, 4, 8];
+        let mut total_capacities = Vec::new();
+
+        for bins in bin_counts {
+            let holder = Arc::new(BucketMapHolder::new(bins, &config, 1));
+            let mut total_capacity = 0;
+
+            for bin in 0..bins {
+                let accounts_index =
+                    InMemAccountsIndex::<u64, u64>::new(&holder, bin, num_initial_accounts);
+                total_capacity += accounts_index.map_internal.read().unwrap().capacity();
+            }
+
+            total_capacities.push(total_capacity);
+
+            if num_initial_accounts.is_some() {
+                assert!(total_capacity > 0);
+            } else {
+                assert_eq!(total_capacity, 0);
+            }
+        }
+
+        // Verify that total capacity across all bins is the same
+        // regardless of bin count
```

Comment:
I suppose this is true because bin counts were chosen all as 2^n, right? Maybe add a comment about that, or possibly we can check a scenario where num_accounts / bin_count falls into a different range of hashmap sizing...

I'm also fine with deleting this test section, doesn't seem to be too important to check relationship of capacities for different scenarios

Reply:
> I suppose this is true because bin counts were chosen all as 2^n, right? 

yes. https://github.com/anza-xyz/agave/pull/8705#discussion_r2475739366 shows the math proof for n=2^k.